### PR TITLE
[video cleanup] clean up multi-GPU video predictor and notebooks; remove dependencies on "CUDA_VISIBLE_DEVICES" in multi-GPU predictor and use `torch.cuda.set_device` instead 

### PR DESCRIPTION
**Pull request by ronghanghu**
(this PR a piece in the [video predictor refactoring stack](https://github.com/facebookresearch/sam3/pulls?q=is%3Apr+label%3Avideo_predictor))

In this PR, we do a major refactoring of the multi-GPU predictor to remove dependencies on "CUDA_VISIBLE_DEVICES" in multi-GPU predictor and use `torch.cuda.set_device` instead.

After this change, the user no longer need to set "CUDA_VISIBLE_DEVICES" explicitly, and is allowed to import torch (or any other modules that uses torch) before building the predictor, which removes a lot of import ordering errors.

We also move both the single-GPU predictor and the multi-GPU predictor into the same file `sam3_video_predictor.py`

### CODE REVIEW

**Comment by ronghanghu** [comment]
(self-stamping this PR given that it only touches the multi-GPU predictor)
